src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -1,138 +1,433 @@
+          setItemsList(items);
+        }
+      }, [listOverflow, items]);
+
+      // Separate effect for handling listOverflow "expand"
+      React.useEffect(() => {
+        if (listOverflow === "expand") {
+          setListHeight("");
+          if (listItemsCount === defaultItemsVisible) {
+            displayDefaultItems();
+          } else {
+            setItemsList(items);
+            setViewAllLabel("View less");
+          }
+        }
+      }, [listOverflow, listItemsCount, items, defaultItemsVisible]);
+
+      // Sets the selected items count on the menu button.
+      let getSelectedItemsCount: any;
+      if (selectedItems[id]?.items.length > 0) {
+        getSelectedItemsCount = !!(selectedItems[id]?.items.length || 0);
+      }
+
+      const styles = useMultiStyleConfig("MultiSelect", {
         width,
+        isBlockElement,
+        isOpen,
+        hasSelectedItems: getSelectedItemsCount,
+      });
+
+      const isChecked = (multiSelectId: string, itemId: string): boolean => {
+        if (selectedItems[multiSelectId]) {
+          return !!selectedItems[multiSelectId].items.find(
+            (selectedItemId: string) => selectedItemId === itemId
+          );
+        }
+        return false;
+      };
+
+      // isAllChecked defines the isChecked status of parent checkboxes. If all child items are selected, it will turn true, otherwise it returns false.
+      // This prop is only passed to parent options.
+      const isAllChecked = (
+        multiSelectId: string,
+        item: MultiSelectItem
+      ): boolean => {
+        let childIds: string[] = item.children.map((childItem) => childItem.id);
+        if (selectedItems[multiSelectId] !== undefined) {
+          return childIds.every(
+            (childItem) =>
+              selectedItems[multiSelectId].items.includes(childItem) &&
+              !item.children.find((c) => c.id === childItem)?.isDisabled
+          );
+        }
+        return false;
+      };
+
+      // isInteterminate will return true if some child items of the parent item are selected. This prop is only passed to parent options.
+      const isIndeterminate = (
+        multiSelectId: string,
+        item: MultiSelectItem
+      ): boolean => {
+        let childIds: string[] = item.children.map((childItem) => childItem.id);
+        if (
+          selectedItems[multiSelectId] !== undefined &&
+          childIds.length > 0 &&
+          childIds.some(
+            (childItem) =>
+              selectedItems[multiSelectId].items.includes(childItem) &&
+              !item.children.find((c) => c.id === childItem)?.isDisabled
+          )
+        ) {
+          return !isAllChecked(multiSelectId, item);
+        }
+        return false;
+      };
+
+      const updateDisabledState = (
+        multiSelectId: string,
+        item: MultiSelectItem
+      ): boolean => {
+        // Update isDisabled for the parent item
+        item.isDisabled = isAllChecked(multiSelectId, item);
```

Comment:
What is this meant to do? `isAllChecked` is checking if the children are checked and not disabled. This value then gets overridden by the if/else statement below.